### ISSUE TITLE
Use test set as validation set

### ISSUE THREAD

**Issue by TohnoYukine** [yashk2810/MNIST-Keras#3]
The way you use test set is invalid. You use test set to show that your training is complete, that's incorrect.
Usually we separate a small portion, say 1/10, as validation set, then we do the training and observe error on training set and validation set. 
Error on training set usually decreases, but with increased sweeps/epoch on training set, error on validation set goes down then slightly up, indicating the model's over-fitting over training set. That is when you should stop training.
The test set is only used to examine the performance of the model when it is trained. Beware! Sweeps/Epoch is a hyper-parameter you tune for the model. You only use validation set to tune it.
